python/lsst/obs/base/cameraMapper.py

```diff
@@ -1062,20 +1064,75 @@ def _standardizeExposure(self, mapping, item, dataId, filter=True,
             The standardized Exposure.
         """
         try:
-            item = exposureFromImage(item, dataId, mapper=self, logger=self.log, setVisitInfo=setVisitInfo)
+            exposure = exposureFromImage(item, dataId, mapper=self, logger=self.log,
+                                         setVisitInfo=setVisitInfo)
         except Exception as e:
             self.log.error("Could not turn item=%r into an exposure: %s" % (repr(item), e))
             raise
 
         if mapping.level.lower() == "amp":
-            self._setAmpDetector(item, dataId, trimmed)
+            self._setAmpDetector(exposure, dataId, trimmed)
         elif mapping.level.lower() == "ccd":
-            self._setCcdDetector(item, dataId, trimmed)
+            self._setCcdDetector(exposure, dataId, trimmed)
+
+        # We can only create a WCS if it doesn't already have one and
+        # we have either a VisitInfo or exposure metadata.
+        if exposure.getWcs() is None and \
+           (exposure.getInfo().getVisitInfo() is not None or exposure.getMetadata().toDict() != {}):
+            self._createInitialSkyWcs(exposure)
 
         if filter:
-            self._setFilter(mapping, item, dataId)
+            self._setFilter(mapping, exposure, dataId)
 
-        return item
+        return exposure
+
+    def _createSkyWcsFromMetadata(self, exposure):
+        """Create a SkyWcs from the FITS header metadata in an Exposure.
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+            The exposure to get metadata from, and attach the SkyWcs to.
+        """
+        metadata = exposure.getMetadata()
+        try:
+            wcs = afwGeom.makeSkyWcs(metadata, strip=True)
+            exposure.setWcs(wcs)
+        except pexExcept.TypeError as e:
+            # See DM-14372 for why this is debug and not warn (e.g. calib files without wcs metadata).
+            self.log.debug("wcs set to None; missing information found in metadata to create a valid wcs:"
+                           " %s", e.args[0])
+        # ensure any WCS values stripped from the metadata are removed in the exposure
+        exposure.setMetadata(metadata)
+
+    def _createInitialSkyWcs(self, exposure):
+        """Create a SkyWcs from the boresight and camera geometry.
+
+        If the boresight or camera geometry do not support this method of
+        WCS creation, this falls back on the header metadata-based version
+        (typically a purely linear FITS crval/crpix/cdmatrix WCS).
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+            The exposure to get data from, and attach the SkyWcs to.
+        """
+        self._createSkyWcsFromMetadata(exposure)
```

Comment:
Would it make sense to have this in the `except` instead of up here? If it takes nonzero time and is not ultimately used in some/most cases, I wouldn't think we need to spend time computing it.

Reply:
It takes negligible time. The reason I do it first is to ensure any WCS-related keys are always stripped from the metadata. I've added a comment to that effect.